### PR TITLE
Restrict parse tags function

### PR DESCRIPTION
**Pull request by soloplxya**
- Restrict the parse_tags function in `ParserUtil.java` to only accept one tag from the user. 
- Update dependencies. 
- Update documentation.
- Update documentation references of `AddressBook` to `WoofAreYou`

### CODE REVIEW

**Review by Tiffanylin21 — COMMENTED** [review]
lgtm! Nice work :)

File: src/main/java/seedu/address/logic/commands/EditCommand.java [hunk 1]
```diff
@@ -43,7 +43,7 @@
             + "[" + PREFIX_PHONE + "PHONE] "
             + "[" + PREFIX_OWNER_NAME + "OWNERNAME] "
             + "[" + PREFIX_ADDRESS + "ADDRESS] "
-            + "[" + PREFIX_TAG + "TAG]...\n"
+            + "[" + PREFIX_TAG + "BREED]"
```
[the remark this answers is not in the document]
Comment (soloplxya) [reply src/main/java/seedu/address/logic/commands/EditCommand.java]: yep thanks for pointing out!

File: src/test/java/seedu/address/logic/parser/EditCommandParserTest.java [hunk 2]
```diff
@@ -114,11 +114,11 @@ public void parse_invalidValue_failure() {
     public void parse_allFieldsSpecified_success() {
         Index targetIndex = INDEX_SECOND_PET;
         String userInput = targetIndex.getOneBased() + PHONE_DESC_BOB + TAG_DESC_HUSBAND
-                + OWNER_NAME_DESC_AMY + ADDRESS_DESC_AMY + NAME_DESC_AMY + TAG_DESC_FRIEND;
+                + OWNER_NAME_DESC_AMY + ADDRESS_DESC_AMY + NAME_DESC_AMY;
 
         EditCommand.EditPetDescriptor descriptor = new EditPetDescriptorBuilder().withName(VALID_NAME_AMY)
                 .withPhone(VALID_PHONE_BOB).withOwnerName(VALID_OWNER_NAME_AMY).withAddress(VALID_ADDRESS_AMY)
-                .withTags(VALID_TAG_HUSBAND, VALID_TAG_FRIEND).build();
+                .withTags(VALID_TAG_HUSBAND).build();
```
[the remark this answers is not in the document]
Comment (soloplxya) [reply src/test/java/seedu/address/logic/parser/EditCommandParserTest.java:121]: Sounds plausible! What do the rest think?

[the remark this answers is not in the document]
Comment (tobihy) [reply src/test/java/seedu/address/logic/parser/EditCommandParserTest.java:121]: Sounds good to me

**Pull request merged**